### ISSUE TITLE
How to initiate "USER_PASSWORD" sign in flow in android.

### ISSUE THREAD

**Issue by sergiotorresperez** [aws-amplify/aws-sdk-android#1778]
We are trying to migrate the cognito user pool in our system.
To do this, we do need to use "USER_PASSWORD" AuthenticationType, as explained [here](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-dg.pdf): 

> Your app sends the username and password to Amazon Cognito. If your app has a native sign-in
> UI and uses the Cognito Identity Provider SDK, your app must use the USER_PASSWORD_AUTH
> flow, in which the SDK sends the password to the server (your app must not use the
> default USER_SRP_AUTH flow since the SDK does not send the password to the server
> in the SRP authentication flow). The USER_PASSWORD_AUTH flow is enabled by setting
> AuthenticationDetails.authenticationType to "USER_PASSWORD".

The [Amplitude documentation](https://docs.aws.amazon.com/cognito/index.html) does not explain how to do this.

[This thread ](https://github.com/aws-amplify/aws-sdk-android/issues/1051) states we need to use 
`android-sdk-cognitoidentityprovider` SDK in order to specify the `AuthenticationType`.

1. - Could you please refer me to one example or code snipped on how to implement this use case using the aforementioned SDK? I can't understand how to do this with the javadoc documentation referred - The javadocs explain how to build one `AuthenticationDetails` object with the desired authentication type, but I don't know what to do with it. 
How could I effectively start the authentication flow with the manually constructed `AuthenticationDetails`?

2. - Our app is currently using the Amplitude SDK extensively for sign-in and authentication features. 
If we use `android-sdk-cognitoidentityprovider` to sign in with  "USER_PASSWORD", will the existing code be compatible with it? 
For example, will the app be able to get the id tokens using `AwsMobileClient.getTokens()`, will the automatic refreshment of expired tokens work, and will the "reset forgotten password" use case work? 

**Which AWS Services are you utilizing?**

com.amazonaws:aws-android-sdk-mobile-client:2.16.10
com.amazonaws:aws-android-sdk-s3:2.16.10
com.amazonaws:aws-android-sdk-cognitoauth:2.16.10

**Environment(please complete the following information):**
 - SDK Version: [e.g. 2.6.25]
2.16.10

**Device Information (please complete the following information):**
 - Device: Any android phone
 - Android Version: minSdkVersion is API 26
 - Specific to simulators: No.

**Comment by OsmarICancino** [comment]
Is there any answer to this issue ?, I'm going through the same problem.

It is really awkward not being able to correctly configure the `AuthenticationType`, there is no reason for it to be specified in `awsconfiguration.json` =/

**Comment by dieehard** [comment]
> * How could I effectively start the authentication flow with the manually constructed `AuthenticationDetails`

```
AuthenticationDetails authenticationDetails = new AuthenticationDetails("username", "password", null);
authenticationDetails.setAuthenticationType(CognitoServiceConstants.CHLG_TYPE_USER_PASSWORD);

AppHelper
	.getPool().getUser(username)
	.initiateUserAuthentication(authenticationDetails, authenticationHandler, true)
	.run();
```

**Comment by richardmcclellan** [comment]
This issue has been fixed in [2.25.0](https://github.com/aws-amplify/aws-sdk-android/releases/tag/release_v2.25.0)
